### PR TITLE
Track first ad loaded

### PR DESCRIPTION
**Pull request by leggsimon**
Adding an oTracking event to gather `firstAdLoaded` performance mark. The `firstAdLoaded` mark was previously being very rarely sent using `n-instrumentation` due to the `onLoad` event happening before the first ad was being loaded.

cc/ @adgad @commuterjoy @ironsidevsquincy @wheresrhys

### CODE REVIEW

File: ads/index.js [hunk 1]
```diff
@@ -73,6 +74,9 @@ function onAdsComplete (event) {
 			utils.log.info('Ad loaded in slot', event);
 			if (slotsRendered === 0) {
 				perfMark('firstAdLoaded');
+				if (!/spoor-id=0/.test(document.cookie)) {
```
Comment (adgad) [inline ads/index.js]: Should we use a different number here? I have a feeling the poor people with spoor-ids starting with 0 are getting a raw deal with all our sampling.

File: ads/index.js [hunk 2]
```diff
@@ -87,6 +91,24 @@ function onAdsComplete (event) {
 	}
 }
 
+function sendAdLoadedTrackingEvent() {
+	const marks = performance.getEntriesByType ?
+		performance.getEntriesByType('mark')
```
Comment (adgad) [inline ads/index.js]: I think you can also do `getEntriesByName` here, and saving having to look through the array

**Comment by leggsimon** [comment]
💭  Just thinking, I guess we'll now need to ensure that n-instrumentation doesn't send the `firstAdLoaded` performance mark if it does by any chance have it before it sends them all to beacon.

**Comment by wheresrhys** [comment]
why have a separate event rather than delay the main navigation timing event?

File: ads/index.js (hunk 2 above)
Comment (adgad) [inline ads/index.js]: Maybe we could also add an offset from domContentLoaded?

File: ads/index.js (hunk 1 above)
Comment (leggsimon) [inline ads/index.js]: wouldn't we want all our performance marks to come from the same sample though?

**Comment by leggsimon** [comment]
@wheresrhys as @ironsidevsquincy said yesterday, you either risk not having the data if someone navigates away from the page before 5000ms (for example) or you bias your data towards users that stay on a page for 5000ms

File: ads/index.js (hunk 1 above)
Comment (adgad) [inline ads/index.js]: I don't think it really matters, since we can't tie the two events up nicely anyway

**Comment by wheresrhys** [comment]
I'd say the risk is overstated, but don't feel that strongly

File: ads/index.js [hunk 3]
```diff
@@ -87,6 +91,23 @@ function onAdsComplete (event) {
...
+	const marks = performance.getEntriesByType ?
+		performance.getEntriesByName('firstAdLoaded')
+			.reduce((marks, mark) => {
+				marks[mark.name] = Math.round(mark.startTime);
+				return marks;
+			}, {}) :
+		{};
+
+	broadcast('oTracking.event', {
+		category: 'page-load',
```
Comment (commuterjoy) [inline ads/index.js]: `category: 'ads', action: 'first-load'` is probably more obvious, as you already have `ads:blocked`

**Comment by adgad** [comment]
:+1

File: ads/index.js (hunk 1 above)
Comment (leggsimon) [inline ads/index.js]: Turns out there's a mistake here and actually the "poor people with spoor-ids starting with 0" are specifiacally *not* getting targeted. Fix [here](https://github.com/Financial-Times/n-ui/pull/104)

**Pull request merged**